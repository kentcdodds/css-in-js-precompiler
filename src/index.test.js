import cssParser from 'css'
import stripIndent from 'strip-indent'

const precompile = require('./')

test('precompiles css-in-js source code', () => {
  const result = precompile({
    source: stripIndent(
      `
        glamorous(
          {
            fontSize: 20,
            fontWeight: 'normal',
          },
          {
            margin: 20,
            ':hover': {
              margin: 10,
            },
          },
        )
        someOtherCall({fontSize: 30})
      `,
    ).trim(),
  })
  const {code, css} = result
  const formattedCSS = cssParser.stringify(cssParser.parse(css))
  expect(code).toMatchSnapshot('1. sanity test code')
  expect(formattedCSS).toMatchSnapshot('2. sanity test css')
  // we send along everything that babel gives us
  // and we don't really care to snapshot that
  // but it'd be nice to know if one of those changes
  // in the future :)
  expect(Object.keys(result)).toEqual([
    'metadata',
    'options',
    'ignored',
    'code',
    'ast',
    'map',
    'css',
  ])
})
