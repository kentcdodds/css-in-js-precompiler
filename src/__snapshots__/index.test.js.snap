// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. sanity test code 1`] = `
"glamorous('css-1i7gko5', 'css-oh2yv');
someOtherCall({ fontSize: 30 });"
`;

exports[`2. sanity test css 1`] = `
".css-1i7gko5,
[data-css-1i7gko5] {
  font-size: 20px;
  font-weight: normal;
}

.css-oh2yv,
[data-css-oh2yv] {
  margin: 20px;
}

.css-oh2yv:hover,
[data-css-oh2yv]:hover {
  margin: 10px;
}"
`;
